gqlalchemy/memgraph.py

```diff
@@ -318,12 +364,22 @@ def load_relationship_with_start_node_and_end_node(
 
     def save_relationship(self, relationship: Relationship) -> Optional[Relationship]:
         if relationship._id is not None:
-            self.save_relationship_with_id(relationship)
+            result = self.save_relationship_with_id(relationship)
         elif relationship._start_node_id is not None and relationship._end_node_id is not None:
-            return self.create_relationship(relationship)
+            result = self.create_relationship(relationship)
         else:
             raise GQLAlchemyError("Can't create a relationship without start_node_id and end_node_id.")
 
+        if self.on_disk_db:
+            for field in relationship.__fields__:
+                value = getattr(relationship, field, None)
+                if value is not None:
+                    if "on_disk" in relationship.__fields__[field].field_info.extra:
+                        self.on_disk_db.save_relationship_property(result._id, field, value)
+                        setattr(result, field, value)
```

Comment:
What if `on_disk_db` is `None` and `on_disk` field is `True`, will that raise an error?

Reply:
if `on_disk_db` is None, that will fail the first if on line 373. But I can make it so it says `if self.on_disk_db is not None`, that is better practice. Or did you have something else in mind?

Reply:
```suggestion
        if self.on_disk_db is not None:
            for field in relationship.__fields__:
                value = getattr(relationship, field, None)
                if value is not None:
                    if "on_disk" in relationship.__fields__[field].field_info.extra:
                        self.on_disk_db.save_relationship_property(result._id, field, value)
                        setattr(result, field, value)
```

Reply:
I think I had something else in mind, what if there is an on_disk prop, but on_disk storage is not defined? That has to fail because it's not good if something the user wanted to be on_disk ends up in-memory. In other words, is it possible to miss-use this API and get inconsistent behavior? NOTE: I didn't check the code deeply. I'm just reasoning based on the small part of the code here.

Reply:
Oh, I think I understand what you're asking. There is undefined behaviour, yes. If the user specifies the property that should go on disk, but he didn't provide an `on_disk_db`, those properties would get dropped. We can throw a `GQLAlchemyError` in those instances upon class init, e.g.:
```
db = Memgraph()

class User(Node):
    huge_string: str = Field(on_disk_db=True, db=db)
```

This should throw an error:
```python
`db` object in line 4, file `tmp.py` doesn't have `on_disk_db` specified.
    huge_string: str = Field(on_disk_db=True, db=db)
    
Add an on_disk_db like this:
from gqlalchemy import Memgraph, SQLOnDiskStorage
db = Memgraph()
SQLOnDiskStorage(db)
```